delivery_purchase_label/models/purchase_order.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import SUPERUSER_ID, _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    delivery_label_picking_id = fields.Many2one(
+        "stock.picking",
+        "Delivery Label Picking",
+        store=True,
+        readonly=True,
+    )
+
+    @api.model
+    def _states_to_generate_delivery_label(self):
+        """Labels will be (re)generated only if the PO is in one of these states."""
+        return ["draft", "sent"]
+
+    @api.model
+    def _mail_templates_to_not_attach_labels(self):
+        return self.env.ref("purchase.email_template_edi_purchase_reminder").ids
+
+    def action_rfq_send(self):
+        self.ensure_one()
+        self._generate_purchase_delivery_label()
+        return super().action_rfq_send()
+
+    def _is_valid_for_vendor_labels(self):
+        self.ensure_one()
+        if self.state not in self._states_to_generate_delivery_label():
+            return
+        if not self.dest_address_id:
+            return False
+        if not any(
+            product.type in ["product", "consu"]
+            for product in self.order_line.product_id
+        ):
+            return False
+        return True
+
+    def _generate_purchase_delivery_label(self):
+        """Create a transfer to generate the carrier labels."""
+        self.ensure_one()
+        if not self._is_valid_for_vendor_labels():
+            return
+        # Find the carrier that will be used
+        carrier = self.partner_id.purchase_delivery_carrier_id
+        if not carrier.purchase_label_picking_type:
+            return
```

Comment:
Is it needed to check the `purchase_label_picking_type` or can we assume that if the `purchase_delivery_carrier_id` is set on the partner we are just using the created operation type from the data file. 
```suggestion
        if not carrier:
            return
```

Reply:
It's better to configure than assume anything